### PR TITLE
Round numbers intelligently in domain data.

### LINKED ISSUE #313: Feature: Rounding in Participant Domain Data
## Feature Details
Apply heuristics for data display in `mod_ParticipantDomain_Server()`. For example, with this:
- Metric = Serious Adverse Event Rate
- Site = 0X109
- Participant = 611
- Metric Summary = Lab

Result has many digits in some cases. That one's a good test, though, because some values are integers.

## Additional Comments
Do we set a standard rounding, or display in scientific notation, or some other heuristic?

### PR DESCRIPTION
**Pull request by jonthegeek**
## Overview
Apply a function to domain gt's (pariticipant tab) to automatically round non-integer numbers in a way that should be aesthetically pleasing.

## Test Notes/Sample Code
Load the sample app, and type "611" in the participant drop-down to choose "0611". Load their various domain tables, noting the "result" column of "Labs" in particular (but also that other columns don't get messed with in other reports). The goal here is to be generic, since I don't know what data might be there in real usage.

## Connected Issues

- Closes #313

### CODE REVIEW

File: R/mod_ParticipantDomain_Server.R [hunk 1]
```diff
@@ -32,7 +37,8 @@ mod_ParticipantDomain_Server <- function(id, rctv_lData, rctv_strName) {
           out_gtInteractive(selection_mode = "multiple") %>%
           gt::cols_label(
             .list = gsm::MakeParamLabelsList(colnames(df))
-          )
+          ) %>%
+          out_gtSmartFmtNumbers(intMaxDecimals = 10L)
```
Comment (lauramaxwell) [inline R/mod_ParticipantDomain_Server.R:41]: Nothing to change now, but should we ask users about the degree of precision they'll need here, or just default to "highly-precise"?

**Review by lauramaxwell — APPROVED** [review]
looks good! just a couple little comments, but nothing to hold this up!

File: R/mod_ParticipantDomain_Server.R (hunk 1 above)
Comment (jonthegeek) [reply R/mod_ParticipantDomain_Server.R:41]: Yeah, I'm definitely thinking we'll want to tune this eventually. The example I gave ends up rounding to 2 decimal places, but I'm not sure what real data will look like. I figured starting with too much and then decreasing if anything is bad is better than cutting off data.

Comment (lauramaxwell) [reply R/mod_ParticipantDomain_Server.R:41]: participant labs was getting rounded to 10, so thats what made me poke around at what the defaults were. nbd, but maybe something to think about once real data is being used as you said.
image

Comment (jonthegeek) [reply R/mod_ParticipantDomain_Server.R:41]: Doh! That's the report I was specifically looking at, but evidently not that site! Interesting!

**Pull request merged**